tests/test_shadow_fork.py

```
@@ -72,30 +77,44 @@ def test_create_fails_if_already_active(self, tmp_path):
+        """Test that create falls back to valid ref if base branch doesn't exist."""
         fork = ShadowFork(repo_root=tmp_path)
+        caplog.set_level(logging.INFO)
 
-        with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
+        # Mock _resolve_base_ref to return a fallback ("HEAD")
+        with patch.object(fork, "_resolve_base_ref", return_value="HEAD") as mock_resolve:
```

Ruff (0.14.14)

[error] Local variable `mock_resolve` is assigned to but never used

Remove assignment to unused variable `mock_resolve`

(F841)